oci/layout/oci_dest.go

```diff
@@ -302,6 +305,33 @@ func (d *ociImageDestination) Commit(context.Context, types.UnparsedImage) error
 	return os.WriteFile(d.ref.indexPath(), indexJSON, 0644)
 }
 
+// PutBlobFromLocalFile arranges the data from path to be used as blob with digest.
+// It computes, and returns, the digest and size of the used file.
+//
+// This function can be used instead of dest.PutBlob() where the ImageDestination requires PutBlob() to be called.
+func PutBlobFromLocalFile(ctx context.Context, dest types.ImageDestination, file string) (digest.Digest, int64, error) {
```

Comment:
I think we need a means to configure the mime type of the blob to serve artifacts.  Going forward, I anticipate users to create their custom layer types for LLM etc.

Reply:
The way the `ImageDestination` API works, `PutBlob` doesn’t typically need to worry about the MIME type: the MIME type comes via `PutManifest`.

While it is true that the `PutBlobWithOptions` gets a full `types.BlobInfo`, and https://github.com/containers/image/blob/0c42f2fe704c9eb84a0b91f4f0363c9919a677db/copy/single.go#L852 forwards the MIME type value into the layer copy code, there are things like https://github.com/containers/image/blob/0c42f2fe704c9eb84a0b91f4f0363c9919a677db/copy/compression.go#L168 just dropping that; so, transports can’t rely on a MIME type being present (~instead, we have `PutBlobOptions.IsConfig).

Also, _specifically in the OCI transport_, the `PutBlob*` code really doesn’t care about the MIME type. It is going to create a sha256-named file, and as far as the layout definition is concerned, it’s perfectly valid to refer to a single blob in two different manifests using two different MIME types. So, for this _OCI-transport-specific_ function, I don’t think adding a MIME type is right.